drivers/adc/ad4080/ad4080.c

```diff
@@ -58,18 +59,95 @@
  * @param reg_val - The data to be written.
  * @return 0 in case of success, negative error code otherwise.
  */
-int ad4080_write(struct ad4080_dev *dev, uint16_t reg_addr, uint8_t reg_val)
+int ad4080_write(struct ad4080_dev *dev, uint32_t reg_addr,
+		 uint32_t reg_val)
 {
-	uint8_t buff[3];
+	int ret;
+	uint32_t reg_size = 0;
+	uint8_t buff[6];
+	uint8_t buf_size = 0;
+	uint8_t address_increment = 1;
+	uint32_t i;
 
 	if (!dev)
 		return -EINVAL;
 
+	reg_size = AD4080_ADDR(reg_addr);
+
+	/* Check addr asc/desc and based on that set the base address and
+	 * address increment */
+	if ((!dev->addr_asc) & (reg_size > 1)) {
+		address_increment = -1;
+		reg_addr += reg_size - 1; /* Switch to address of last byte */
+	}
+
+	/* In this case, only a single transaction is required. */
+	if (reg_size == 1 || dev->strict_reg || !dev->single_instr) {
+
+		if (!dev->short_instr) {
+			/* Fill up the buffer execute and return */
 	buff[0] = no_os_field_get(BYTE_ADDR_H, reg_addr);
 	buff[1] = no_os_field_get(BYTE_ADDR_L, reg_addr);
-	buff[2] = reg_val;
+			buf_size = LONG_INSTR_LENGTH;
+		} else {
+			buff[0] = no_os_field_get(BYTE_ADDR_L, reg_addr);
+			buf_size = SHORT_INSTR_LENGTH;
+		}
+
+		for (i = 0; i < reg_size; i++) {
+			buff[buf_size + i - 1] = dev->addr_asc ?
```

Comment:
i think it is easier to understand if you use standard if else operations instead of ternary operators.